src/main/java/juja/microservices/teams/service/TeamService.java

```diff
@@ -3,54 +3,84 @@
+import juja.microservices.teams.exceptions.UserNotInTeamException;
 import org.springframework.stereotype.Service;
 
 import javax.inject.Inject;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 /**
  * @author Andrii.Sidun
+ * @author Ivan Shapovalov
  */
-
 @Service
+@Slf4j
 public class TeamService {
-
-    private final Logger logger = LoggerFactory.getLogger(this.getClass());
-
     @Inject
     private TeamRepository teamRepository;
 
-    public String addTeam(TeamRequest teamRequest){
-        logger.debug("Start TeamService.addTeam. Team: {}", teamRequest);
-        if (isAnyBodyInOtherTeam(teamRequest)){
-            logger.warn("Some user(s) exists in a team");
-            throw new TeamUserExistsException("Some user(s) exists in a team");
+    public Team addTeam(TeamRequest teamRequest) {
+        log.debug("Started 'addTeam' TeamRequest: {}", teamRequest);
+        Set<String> usersInTeams = usersInCurrentTeams(teamRequest);
+        if (usersInTeams.size() > 0) {
+            log.warn("User(s) '{}' exists in a another teams",usersInTeams);
+            throw new UserExistsException(String.format("User(s) '%s' exists in a another teams", usersInTeams.toString()));
         }
         Team team = mappingRequestToTeam(teamRequest);
-        String newTeamId = teamRepository.add(team);
-        logger.info("Added new Team with parameters '{}'", team);
-        logger.debug("Finish TeamService.addTeam. newTeamId: {}", newTeamId);
-        return newTeamId;
+        log.debug("Started 'Save team '{}'", team);
+        teamRepository.saveTeam(team);
+        log.info("Finished 'Save team' '{}'", team.getId());
+        log.debug("Finished 'Save team '{}'", team);
+        return team;
     }
 
     private Team mappingRequestToTeam(TeamRequest teamRequest) {
-        return new Team(teamRequest.getFrom(), teamRequest.getUuidOne(), teamRequest.getUuidTwo(),
-                teamRequest.getUuidThree(), teamRequest.getUuidFour());
+        return new Team(teamRequest.getMembers());
     }
 
-    private boolean isAnyBodyInOtherTeam(TeamRequest team){
-        logger.debug("Start isAnyBodyInOtherTeam()");
-        boolean result = false;
-        if (teamRepository.isUserInOtherTeam(team.getUuidOne()) ||
-                teamRepository.isUserInOtherTeam(team.getUuidTwo()) ||
-                teamRepository.isUserInOtherTeam(team.getUuidThree()) ||
-                teamRepository.isUserInOtherTeam(team.getUuidFour())) {
-            result = true;
+    private Set<String> usersInCurrentTeams(TeamRequest teamRequest) {
+        log.debug("Started 'usersInCurrentTeams' with teamRequest '{}'",teamRequest);
+        Set<String> usersInTeams = new HashSet<>();
+        teamRequest.getMembers()
+                .forEach(uuid -> {
+                    List<Team> teams = teamRepository.getUserTeams(uuid);
+                    if (teams.size() != 0) {
```

Comment:
Can we find another solution for such request - to do one query to the database with parameter like `any`?